### PR TITLE
Update new button component to use the new tokens

### PR DESCRIPTION
**Pull request by salamaashoush**

### CODE REVIEW

File: packages/libs/react-ui/src/components/Button/NewButton.css.ts [hunk 1]
```diff
@@ -13,128 +11,206 @@ const outlineColor = createVar();
 const backgroundColor = createVar();
 const color = createVar();
```
Comment (isaozler) [inline packages/libs/react-ui/src/components/Button/NewButton.css.ts:12]: missing the iconColor

**Review by ferreroltd — APPROVED** [review]

**Review by eileenmguo — COMMENTED** [review]
It looks like the `isCompact` arg isn't doing anything on storybook. 

Before this PR is merged, I think the old button should be removed and replaced with this new one so that the apps are updated as well 🎄

**Comment by salamaashoush** [comment]
> It looks like the `isCompact` arg isn't doing anything on storybook.
> 
> Before this PR is merged, I think the old button should be removed and replaced with this new one so that the apps are updated as well 🎄

Yes there was cascading issues for the padding since atoms have low priority than styles in vanilla extract

to this PR small and easy to review I will create a separate MR to update the apps and remove the old button

File: packages/libs/react-ui/src/components/Button/NewButton.css.ts (hunk 1 above)
Comment (salamaashoush) [reply packages/libs/react-ui/src/components/Button/NewButton.css.ts]: done

**Review by isaozler — APPROVED** [review]

**Pull request closed without merging**